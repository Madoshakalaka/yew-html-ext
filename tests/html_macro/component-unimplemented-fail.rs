struct Unimplemented;

fn compile_fail() {
    ::yew_html_ext::html! { <Unimplemented /> };
}

fn main() {}
