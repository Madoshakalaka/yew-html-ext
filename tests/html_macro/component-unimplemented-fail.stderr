error[E0277]: the trait bound `Unimplemented: yew::Component` is not satisfied
 --> tests/html_macro/component-unimplemented-fail.rs:4:30
  |
4 |     ::yew_html_ext::html! { <Unimplemented /> };
  |                              ^^^^^^^^^^^^^ the trait `yew::Component` is not implemented for `Unimplemented`, which is required by `Unimplemented: BaseComponent`
  |
  = help: the trait `yew::Component` is implemented for `ContextProvider<T>`
  = note: required for `Unimplemented` to implement `BaseComponent`
  = note: this error originates in the macro `::yew_html_ext::html` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0599]: the function or associated item `new` exists for struct `VChild<Unimplemented>`, but its trait bounds were not satisfied
 --> tests/html_macro/component-unimplemented-fail.rs:4:30
  |
1 | struct Unimplemented;
  | -------------------- doesn't satisfy `Unimplemented: BaseComponent` or `Unimplemented: yew::Component`
...
4 |     ::yew_html_ext::html! { <Unimplemented /> };
  |                              ^^^^^^^^^^^^^ function or associated item cannot be called on `VChild<Unimplemented>` due to unsatisfied trait bounds
  |
  = note: the following trait bounds were not satisfied:
          `Unimplemented: yew::Component`
          which is required by `Unimplemented: BaseComponent`
note: the trait `yew::Component` must be implemented
 --> $CARGO/yew-0.21.0/src/html/component/mod.rs
  |
  | pub trait Component: Sized + 'static {
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  = note: this error originates in the macro `::yew_html_ext::html` (in Nightly builds, run with -Z macro-backtrace for more info)
